### PR TITLE
make network functions into objects

### PR DESCRIPTION
**Pull request by Vtrolia**
I believe an object-oriented approach for our network functions would make things more portable and easier so here's my proposal for a base connection class, and client and server objects

### CODE REVIEW

**Review by gallimania — APPROVED** [review]
Great work!

File: networking.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (C) 2018 Garrett Fifer
+# Copyright (C) 2018 Illogical Mathmatical Hypothetical Numerical Lyrical Crew
```
Comment (gallimania) [inline networking.py:1]: Highly approve

**Review by gallimania — COMMENTED** [review]
No other words needed.

**Comment by Vtrolia** [comment]
bug fixes & stability improvements

**Pull request merged**